### PR TITLE
store repository URL in rooted repositories, closes #74

### LINKED ISSUE #74: store repository URL in rooted repositories
URL which was used to fetch a repository should be present in rooted repositories. We can do this by leveraging git `remotes` config.

The result would be:
* For each remote repository, there is a remote with the repository ULID as its remote name.
* We would not create special names for the references like now, ULID would be just the remote name in reference names like `refs/remotes/<ULID>/<branch>`

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

**Comment by smola** [comment]
Depends on https://github.com/src-d/go-git/pull/501 if we want multiple endpoints.

File: git.go [hunk 1]
```diff
@@ -219,6 +220,31 @@ func (r *temporaryRepository) Push(
...
+
+		remote.URLs = mr.Endpoints
+	} else {
+		c.Remotes[id] = &config.RemoteConfig{
+			Name: id,
+			URLs: mr.Endpoints,
+		}
+	}
+
+	return storer.SetConfig(c)
```
Comment (ajnavarro) [inline git.go]: You only need to set config if the remotes changes.

Comment (smola) [inline git.go]: That's already the case ;)

Comment (ajnavarro) [inline git.go]: oops, so true

**Pull request merged**